virtwho/virt/xen/xen.py

```diff
@@ -187,8 +187,9 @@ def _run(self):
                 events = []
 
             if initial or len(events) > 0 or delta > 0:
-                assoc = self.getHostGuestMapping()
-                self._send_data(virt.HostGuestAssociationReport(self.config, assoc))
+                if self.is_registered() is True:
```

Comment:
When I'm looking at times in this log: https://bugzilla.redhat.com/attachment.cgi?id=1228247 ('Interrupted system call': 2016-12-02 00:18:21 and 'Unable to read certificate, system is not registered or you are not root': 2016-12-02 00:18:27) I guess the interrupted system call can be just some coincidence. And further consider this: virt-who at the first time tries to collect host-to-guest mapping and then it tries to send these results to candlepin.